### PR TITLE
fix: Add service support for parsing KServe raw contents

### PR DESCRIPTION
**Pull request by ruivieira**
[RHODS-8277](https://issues.redhat.com/browse/RHODS-8277) / #170 

Add support for raw-encoded KServe protobuf payloads.

### CODE REVIEW

File: explainability-service/src/test/java/org/kie/trustyai/service/endpoints/consumer/ConsumerEndpointRawTest.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.kie.trustyai.service.PayloadProducer.MODEL_A_ID;
+
+/**
+ * Test the {@link ConsumerEndpoint} with raw content payloads.
+ */
+@QuarkusTest
+@TestProfile(BaseTestProfile.class)
+@TestHTTPEndpoint(ConsumerEndpoint.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```
Comment (danielezonca) [inline explainability-service/src/test/java/org/kie/trustyai/service/endpoints/consumer/ConsumerEndpointRawTest.java]: Leftover? I don't see the `@Order` annotation in this class

File: explainability-service/src/test/java/org/kie/trustyai/service/endpoints/consumer/ConsumerEndpointRawTest.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+import org.kie.trustyai.service.payloads.consumer.InferencePartialPayload;
+import org.kie.trustyai.service.payloads.consumer.PartialKind;
+
+import io.quarkus.test.common.http.TestHTTPEndpoint;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.restassured.http.ContentType;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.is;
```
Comment (danielezonca) [inline explainability-service/src/test/java/org/kie/trustyai/service/endpoints/consumer/ConsumerEndpointRawTest.java:33]: I usually prefer to limit the library we use for assertions, is it possible to remove this usage in favour of `jupiter.api.Assertions.*`?

File: explainability-service/src/test/java/org/kie/trustyai/service/payloads/RawConverterUtils.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package org.kie.trustyai.service.payloads;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import com.google.protobuf.ByteString;
+
+public class RawConverterUtils {
```
Comment (danielezonca) [inline explainability-service/src/test/java/org/kie/trustyai/service/payloads/RawConverterUtils.java]: This class seems to be a duplicate of [explainability-connectors/src/test/java/org/kie/trustyai/connectors/kserve/v2/RawConverterUtils.java](https://github.com/trustyai-explainability/trustyai-explainability/blob/d05587d96ae18c36555dcd468482bf116fb5067f/explainability-connectors/src/test/java/org/kie/trustyai/connectors/kserve/v2/RawConverterUtils.java)

File: explainability-connectors/src/main/java/org/kie/trustyai/connectors/kserve/v2/RawConverter.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package org.kie.trustyai.connectors.kserve.v2;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.protobuf.ByteString;
+
+public class RawConverter {
```
Comment (danielezonca) [inline explainability-connectors/src/main/java/org/kie/trustyai/connectors/kserve/v2/RawConverter.java:10]: Is it necessary to have `RawConverter` and `RawConverterUtils`? They look similar in terms of logic

**Review by danielezonca — COMMENTED** [review]
Thanks for the PR :)
Few comments but overall looks good to me.
Can you confirm that this PR replace this other?
- https://github.com/trustyai-explainability/trustyai-explainability/pull/139

File: explainability-service/src/test/java/org/kie/trustyai/service/payloads/RawConverterUtils.java (hunk 3 above)
Comment (ruivieira) [reply explainability-service/src/test/java/org/kie/trustyai/service/payloads/RawConverterUtils.java:8]: Moved into `RawConverter`

File: explainability-connectors/src/main/java/org/kie/trustyai/connectors/kserve/v2/RawConverter.java (hunk 4 above)
Comment (ruivieira) [reply explainability-connectors/src/main/java/org/kie/trustyai/connectors/kserve/v2/RawConverter.java:10]: Moved into `RawConverter`

File: explainability-service/src/test/java/org/kie/trustyai/service/endpoints/consumer/ConsumerEndpointRawTest.java (hunk 2 above)
Comment (ruivieira) [reply explainability-service/src/test/java/org/kie/trustyai/service/endpoints/consumer/ConsumerEndpointRawTest.java:33]: Matchers to be cleaned in a separate PR.

**Comment by ruivieira** [comment]
> Thanks for the PR :) Few comments but overall looks good to me. Can you confirm that this PR replace this other?
> 
>     * [FAI-979 V2: Fallback raw conversions #139](https://github.com/trustyai-explainability/trustyai-explainability/pull/139)

This is my understanding. IIUC That PR was built on top of https://github.com/trustyai-explainability/trustyai-explainability/pull/136, which changed in the meantime.

**Review by danielezonca — APPROVED** [review]
Thanks for the changes, approving :)

File: explainability-connectors/src/main/java/org/kie/trustyai/connectors/kserve/v2/RawConverter.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        final List<Boolean> booleanList = new ArrayList<>();
+        for (byte b : bytes) {
+            for (int i = 7; i >= 0; --i) {
+                booleanList.add((b & (1 << i)) != 0);
+            }
+        }
+        return booleanList;
+    }
+
+    public static List<Integer> toInteger(ByteString raw) {
```
Comment (RobGeada) [inline explainability-connectors/src/main/java/org/kie/trustyai/connectors/kserve/v2/RawConverter.java:23]: This parses `int`s backwards: 1 is being encoded to `72057594037927936`. I think the encoding is little-endian, at least for integers.

**Comment by RobGeada** [comment]
> Thanks for the PR :) Few comments but overall looks good to me. Can you confirm that this PR replace this other?
> 
> * [FAI-979 V2: Fallback raw conversions #139](https://github.com/trustyai-explainability/trustyai-explainability/pull/139)

My fallback conversions PR was just a draft to show the issue + a workaround, was never intended to be a fix. So yes, this PR replaces the other.

**Review by RobGeada — APPROVED** [review]

**Pull request merged**